typings/internalBinding/modules.d.ts

```diff
@@ -1,26 +1,30 @@
 export type PackageType = 'commonjs' | 'module' | 'none'
-export type PackageConfig = {
-  pjsonPath: string
+export type RecognisedPackageConfig = {
+  pjsonPath: URL['pathname']
   exists: boolean
   name?: string
   main?: any
   type: PackageType
   exports?: string | string[] | Record<string, unknown>
   imports?: string | string[] | Record<string, unknown>
 }
+export type FullPackageConfig = RecognisedPackageConfig & {
+  [key: string]: unknown,
+}
 export type SerializedPackageConfig = [
-  PackageConfig['name'],
-  PackageConfig['main'],
-  PackageConfig['type'],
+  RecognisedPackageConfig['name'],
```

Comment:
I disagree: `PackageConfig` sounds like the superset of all config options. That's not what it actually is. The potential for confusion is now much more because there are now both options (subset and superset). This rename removes the ambiguity to preclude confusion.